__tests__/app.test.js

```diff
@@ -197,14 +197,57 @@ describe("/api/reviews", () => {
           });
         });
     });
-    test("reviews are sorted by date in descending order", () => {
+    test("reviews are sorted by date in descending order by default", () => {
       return request(app)
         .get("/api/reviews")
         .expect(200)
         .then(({ body }) => {
           expect(body.reviews).toBeSortedBy("created_at", { descending: true });
         });
     });
+    test("reviews can be selected by category", () =>{
+      return request(app)
+      .get("/api/reviews?category=social deduction")
+      .expect(200)
+      .then(({body}) => {
+        expect(body.reviews.length).toBe(11)
+        body.reviews.forEach(review => {
+          expect(review.category).toBe("social deduction")
+        })
+      })
+    })
+    test("reviews can be sorted by any valid column", () => {
+      return request(app)
+        .get("/api/reviews?sort_by=votes")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.reviews).toBeSortedBy("votes", {descending: true });
+        });
+    });
+    test("if invalid sort column given, returns 400 error and error message", () => {
+      return request(app)
+        .get("/api/reviews?sort_by=hello")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Invalid sort query");
+        });
+    });
+    test("if sort order is given, will sort accordingly", () => {
+      return request(app)
+        .get("/api/reviews?sort_by=votes&order=ASC")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.reviews).toBeSortedBy("votes");
+        });
+    });
+    test("if invalid sort order given, returns 400 error and error message", () => {
```

Comment:
We want to do a test here for each one separately otherwise we don't know which one is sending the error/ if they all will send one

Status 400. invalid sort_by query, e.g. ?sort_by=bananas
Status 400. invalid order query, e.g. ?order=bananas